libs/designer-ui/src/lib/picker/filepickereditor.tsx

```diff
@@ -38,35 +38,67 @@ const calloutProps = { gapSpace: 0 };
 export const FilePickerEditor = ({
   initialValue,
   isLoading = false,
+  type,
+  items,
+  displayValue,
+  // fileFilters,
+  errorDetails,
   editorBlur,
-  onChange,
   pickerCallbacks,
   ...baseEditorProps
 }: FilePickerEditorProps) => {
-  const [value, setValue] = useState(initialValue);
-  const [showPicker, setShowPicker] = useState(false);
-  const [titleSegments] = useState<PickerTitleInfo[] | undefined>();
   const pickerIconId = useId();
   const intl = useIntl();
+  const [selectedItem, setSelectedItem] = useState<any>();
+  const [editorDisplayValue, setEditorDisplayValue] = useState<ValueSegment[]>(
+    displayValue ? [{ id: guid(), value: displayValue, type: ValueSegmentType.LITERAL }] : initialValue
+  );
+  const [pickerDisplayValue, setPickerDisplayValue] = useState<ValueSegment[]>([]);
+  const [showPicker, setShowPicker] = useState(false);
+
+  useEffect(() => {
+    if (displayValue) {
+      setPickerDisplayValue([{ id: guid(), value: displayValue, type: ValueSegmentType.LITERAL }]);
+    }
+  }, [displayValue]);
 
-  const { onFolderNavigation: onFolderNavigated } = pickerCallbacks;
+  const { onFolderNavigation, getFileSourceName, getDisplayValueFromSelectedItem, getValueFromSelectedItem } = pickerCallbacks;
+  const fileSourceName = getFileSourceName?.();
 
-  const openTokenPicker = () => {
+  const [titleSegments, setTitleSegments] = useState<IBreadcrumbItem[]>(getInitialTitleSegments(fileSourceName));
+
+  const openFolderPicker = () => {
     if (!showPicker) {
-      onFolderNavigated?.(/* selectedItem */ undefined);
+      setTitleSegments(getInitialTitleSegments(fileSourceName));
+      onFolderNavigation?.(/* selectedItem */ undefined);
       setShowPicker(true);
     }
   };
 
-  const onValueChange = (newValue: ValueSegment[]): void => {
-    setValue(newValue);
-    // TODO: Make sure to update displayValue and selectedItem in viewModel.
-    onChange?.({ value: newValue, viewModel: { hideErrorMessage: true } });
+  const onFolderNavigated = (selectedItem: any) => {
+    onFolderNavigation?.(selectedItem);
+    const displayValue = getDisplayValueFromSelectedItem?.(selectedItem);
+    setTitleSegments([...titleSegments, { text: displayValue, key: displayValue, onClick: () => onFolderNavigated(selectedItem) }]);
+  };
+
+  const onFolderSelected = (selectedItem: any) => {
+    if (showPicker) {
+      setSelectedItem(selectedItem);
+      setPickerDisplayValue([{ id: guid(), value: selectedItem.Path, type: ValueSegmentType.LITERAL }]);
```

Comment:
Why are we defaulting to Path here? you should be calling getDisplayValueFromSelectedItem to set the value no?